java/Makefile

```diff
@@ -216,13 +216,17 @@ endif
 SEARCH_REPO_URL?=http://search.maven.org/remotecontent?filepath=
 CENTRAL_REPO_URL?=https://repo1.maven.org/maven2/
 
-clean:
+clean: clean-not-downloaded clean-downloaded
+
+clean-not-downloaded:
 	$(AM_V_at)rm -rf include/*
-	$(AM_V_at)rm -rf test-libs/
 	$(AM_V_at)rm -rf $(OUTPUT)
 	$(AM_V_at)rm -rf $(BENCHMARK_OUTPUT)
 	$(AM_V_at)rm -rf $(SAMPLES_OUTPUT)
 
+clean-downloaded:
+	$(AM_V_at)rm -rf test-libs/
```

Comment:
how about $(JAVA_TESTLIB_DIR)
And for bonus points, $(NATIVE_INCLUDE) in clean-not-downloaded